### PR TITLE
MAUI eBook Navigation

### PR DESCRIPTION
**Pull request by michaelstonis**
## Summary

Navigation chapter for .NET MAUI eBook

### CODE REVIEW

**Review by IEvangelist — COMMENTED** [review]
A few nits

File: docs/architecture/maui/navigation.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+The following code example shows the implementation of the `MauiNavigationService.InitializeAsync` method:
+
+```csharp
+public Task InitializeAsync()
+{
+    if (string.IsNullOrEmpty(_settingsService.AuthAccessToken))
+        return NavigateToAsync("//Login");
+    else
+        return NavigateToAsync("//Main/Catalog");
```
[the remark this answers is not in the document]
Comment (michaelstonis) [reply docs/architecture/maui/navigation.md:161]: No problem with that. I like the readability of it, but typically lean on more explicit statements for examples. I can update this for other examples!

**Comment by IEvangelist** [comment]
The same warnings/errors are causing issues here, as mentioned in https://github.com/dotnet/docs/pull/30112#issuecomment-1179278460

**Review by IEvangelist — APPROVED** [review]

File: docs/architecture/maui/navigation.md [hunk 2]
```diff
@@ -1,9 +1,240 @@
...
+The `MauiNavigationService` class, which implements the `INavigationService` interface, is registered as a singleton with the dependency injection container in the `MauiProgram.CreateMauiApp()` method, as demonstrated in the following code example:
+
+```csharp
+mauiAppBuilder.Services.AddSingleton<INavigationService, MauiNavigationService>();;
+```
+
+The `INavigationService` interface can then be resolved by adding it to the constructor of our views and view-models, as demonstrated in the following code example:
+
+```csharp
+public AppShell (INavigationService navigationService)
```
Comment (IEvangelist) [inline docs/architecture/maui/navigation.md:65]:
```suggestion
public AppShell(INavigationService navigationService)
```

File: docs/architecture/maui/navigation.md [hunk 3]
```diff
@@ -1,9 +1,240 @@
...
+Navigation is performed inside view-model classes by invoking one of the `NavigateToAsync` methods, specifying the route path for the page being navigated to, as demonstrated in the following code example:
+
+```csharp
+await NavigationService.NavigateToAsync("//Main");
+```
+
+The following code example shows the `NavigateToAsync` method provided by the `MauiNavigationService` class:
+
+```csharp
+public Task NavigateToAsync (string route, IDictionary<string, object> routeParameters = null)
```
Comment (IEvangelist) [inline docs/architecture/maui/navigation.md:85]:
```suggestion
public Task NavigateToAsync(string route, IDictionary<string, object> routeParameters = null)
```

File: docs/architecture/maui/navigation.md [hunk 4]
```diff
@@ -1,9 +1,240 @@
...
+The method uses an instance of `INavigationService` which is provided the constructor from dependency injection and invokes its `InitializeAsync` method.
+
+The following code example shows the implementation of the `MauiNavigationService.InitializeAsync` method:
+
+```csharp
+public Task InitializeAsync()
+{
+        return NavigateToAsync(string.IsNullOrEmpty(_settingsService.AuthAccessToken)
+            ? "//Login"
+            : "//Main/Catalog";
```
Comment (IEvangelist) [inline docs/architecture/maui/navigation.md:160]:
```suggestion
    return NavigateToAsync(string.IsNullOrEmpty(_settingsService.AuthAccessToken)
        ? "//Login"
        : "//Main/Catalog";
```

**Pull request merged**